lib/src/layer/marker_layer.dart

```diff
@@ -14,51 +14,34 @@ class Anchor {
       : left = _leftOffset(width, alignOpt),
         top = _topOffset(height, alignOpt);
 
-  static double _leftOffset(double width, AnchorAlign alignOpt) {
-    switch (alignOpt) {
-      case AnchorAlign.left:
-        return 0;
-      case AnchorAlign.right:
-        return width;
-      case AnchorAlign.top:
-      case AnchorAlign.bottom:
-      case AnchorAlign.center:
-      default:
-        return width / 2;
-    }
-  }
-
-  static double _topOffset(double height, AnchorAlign alignOpt) {
-    switch (alignOpt) {
-      case AnchorAlign.top:
-        return 0;
-      case AnchorAlign.bottom:
-        return height;
-      case AnchorAlign.left:
-      case AnchorAlign.right:
-      case AnchorAlign.center:
-      default:
-        return height / 2;
-    }
-  }
-
-  factory Anchor.forPos(AnchorPos<dynamic>? pos, double width, double height) {
+  static double _leftOffset(double width, AnchorAlign alignOpt) =>
+      switch (alignOpt) {
+        AnchorAlign.left => 0,
+        AnchorAlign.right => width,
+        _ => width / 2,
+      };
+
+  static double _topOffset(double height, AnchorAlign alignOpt) =>
+      switch (alignOpt) {
+        AnchorAlign.top => 0,
+        AnchorAlign.bottom => height,
+        _ => height / 2,
+      };
+
+  factory Anchor.forPos(AnchorPos? pos, double width, double height) {
     if (pos == null) return Anchor._(width, height, AnchorAlign.none);
-    if (pos.value is AnchorAlign) {
-      return Anchor._(width, height, pos.value as AnchorAlign);
-    }
-    if (pos.value is Anchor) return pos.value as Anchor;
-    throw Exception('Unsupported AnchorPos value type: ${pos.runtimeType}.');
+    if (pos.alignment != null) return Anchor._(width, height, pos.alignment!);
+    if (pos.anchor != null) return pos.anchor!;
```

Comment:
You can use pattern matching for both statements:
```dart
if (pos.alignment case final align?) return Anchor._(width, height, align);
if (pos.anchor case final anchor?) return anchor;
```
This will avoid to use the `!` operator.

Reply:
(Was my change.)
Didn't realise this feature. Not too bad, but I think readability suffers a bit. Happy with it though.

```suggestion
    if (pos.alignment case final align?) return Anchor._(width, height, align);
    if (pos.anchor case final anchor?) return anchor;
```